src/lib/Preview.js

```diff
@@ -782,6 +848,23 @@ class Preview extends EventEmitter {
     }
 
     /**
+     * Enrich file entries with offline links.
+     *
+     * @private
+     * @return {void}
+     */
+    enrichOfflineLinks() {
```

Comment:
Another approach, that will require some refactoring, is to change the networking layer. From the top of my head, it could be:

To show a preview one does `show()` --> `getTokens()` --> `getFile()` --> `loadRep()` --> preview. Other than the networking done inside `pdf.js` or `shaka-player.js` or done by the browser (think `<img src=''url" />`), which is out of scope, everything goes through `util.js`.

Network calls can be refactored out of `util.js` and into a class `Xhr.js`. An override of `Xhr.js`, lets call it `Offline.js` can implement `get()`, `post()` etc...which fetches data out of some local offline cache instead of getting it over the wire. As long as the data returned is in the same format as our V2 API, `preview.js` should consume without knowing where it came from.

`preview.js` on `show()` could initialize its networking layer based on some passed in option or a fallback from a network error.
```js
if (options.offline) {
    this.xhr = new Offline();
} else {
    this.xhr = new Xhr();
}```

and then inside `preview.js` and the viewers, one will use `this.xhr.get()` as opposed to the current `util.get()`

`getFile()` will have to revert back to `Xhr` if the file loaded ends up being a non-document (non-offline) file and call `this.load()` again to repeat the above process and go the online route or error out.

Obviously, this won't solve the prior mentioned issue where the networking is done inside `pdf.js` or `shaka-player.js` to load the actual representation data (unless the offline links being provided are actually URLs that open without a internet connection, like from localhost).